components/BurgerMenuNavCategories.vue

```diff
@@ -0,0 +1,342 @@
+            :aria-expanded="isCategoryOpen(index) ? 'true' : 'false'"
+            :aria-label="
+              isCategoryOpen(index)
+                ? `${$t('burger-menu.close-category')} ${item.name}`
+                : `${$t('burger-menu.open-category')} ${item.name}`
+            "
+            :style="{
+              justifyContent: isCategoryOpen(index)
+                ? 'flex-start'
+                : 'space-between',
+            }"
+            @click="toggleCategory(index)"
+          >
+            <fa
+              v-if="isCategoryOpen(index)"
+              class="burger-menu-nav-list-item-category__come-back"
+              icon="angle-left"
+            />
+            {{ item.name }}
+            <fa
+              v-if="!isCategoryOpen(index)"
+              class="burger-menu-nav-list-item-category__come-back"
+              icon="angle-right"
+            />
+          </button>
+        </div>
+        <div v-show="isCategoryOpen(index)">
+          <burger-menu-nav-sections
+            :sections="item.sections"
+            @close-menu="handleLinkClick"
+          />
+        </div>
+      </div>
+      <div v-else v-show="openDropdownIndex === -1" @click="handleLinkClick">
+        <pix-prismic-link
+          :field="item"
+          class="burger-menu-nav-list-item__link"
+          :class="{ 'blue-link': shouldApplyBlueClass(item.url) }"
+        >
+          <img
+            v-if="item.beforeIcon"
+            :src="`/images/${item.beforeIcon}`"
+            :alt="item.alternativeTextForBeforeIcon"
+            role="presentation"
+            width="22"
+            height="18"
+          />
+          {{ item.name }}
+          <img
+            v-if="item.afterIcon"
+            :src="`/images/${item.afterIcon}`"
+            :alt="item.alternativeTextForAfterIcon"
+            role="presentation"
+            width="22"
+            height="18"
+          />
+        </pix-prismic-link>
+      </div>
+    </li>
+  </ul>
+</template>
+
+<script>
+import BurgerMenuNavSections from '~/components/BurgerMenuNavSections.vue'
+
+export default {
+  name: 'BurgerMenuNavCategories',
+  components: {
+    BurgerMenuNavSections,
+  },
+  props: {
+    items: {
+      type: Array,
+      default: null,
+    },
+    isLocaleSwitcherOpen: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      openDropdownIndex: -1,
+    }
+  },
+  computed: {
+    isLocaleSwitcherOpenLocal: {
+      get() {
+        return this.isLocaleSwitcherOpen
+      },
+      set(value) {
+        if (value) {
+          this.openDropdownIndex = -1
+        }
+      },
+    },
+    navigationLinks() {
+      if (this.items.length === 0) {
+        return null
+      }
+
+      return this.items.reduce((formattedData, currentItem) => {
+        const isCategory =
+          currentItem.categoryName.length > 0 &&
+          currentItem.categoryName[0].text
+
+        if (!isCategory) {
```

Comment:
```suggestion
        const isFirstLevelLink = currentItem.menuSectionTitle.length === 0
        
        if (isFirstLevelLink) {
```

Expliciter le type de contenu sur lequel on va travailler dans le `if` ?

Reply:
ℹ️ On fera une refacto commune avec le menu desktop lors de la montée en Nuxt 3 !

Reply:
Vu ensemble. On refacto sur la montée en nuxt 3